### PR TITLE
Provide hover participant

### PR DESCRIPTION
**Pull request by angelozerr**
Provide hover participant

See https://github.com/redhat-developer/quarkus-ls/issues/229

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

File: microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/java/IJavaDiagnosticsParticipant.java [hunk 1]
```diff
@@ -4,14 +4,15 @@
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v20.html
 *
+* SPDX-License-Identifier: EPL-2.0
```
Comment (xorye) [inline microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/java/IJavaDiagnosticsParticipant.java]: Do we need these headers for this project?

File: microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/java/IJavaHoverParticipant.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	 * @return true if hover must be collected for the given context and false
+	 *         otherwise.
+	 * @throws CoreException
+	 */
+	default boolean isAdaptedForHover(JavaHoverContext context, IProgressMonitor monitor) throws CoreException {
+		return true;
+	}
+
+	/**
+	 * Begin the collection of hover.
```
Comment (xorye) [inline microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/java/IJavaHoverParticipant.java]: Begin hover collection

File: microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/java/IJavaHoverParticipant.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	 * Collect hover according to the context.
+	 * 
+	 * @param context the java hover context
+	 * @param monitor the progress monitor
+	 * @throws CoreException
+	 */
+	Hover collectHover(JavaHoverContext context, IProgressMonitor monitor) throws CoreException;
+
+	/**
+	 * End the collection of hover.
```
Comment (xorye) [inline microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/java/IJavaHoverParticipant.java]: End hover collection

File: microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/java/IJavaDiagnosticsParticipant.java [hunk 4]
```diff
@@ -34,22 +35,44 @@
...
+	 * 
 	 */
 	default boolean isAdaptedForDiagnostics(JavaDiagnosticsContext context, IProgressMonitor monitor)
-			throws JavaModelException {
+			throws CoreException {
 		return true;
 	}
 
+	/**
+	 * Begin the collection of diagnostics.
```
Comment (xorye) [inline microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/java/IJavaDiagnosticsParticipant.java]: Begin diagnostics collection.

File: microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/java/IJavaDiagnosticsParticipant.java [hunk 5]
```diff
@@ -34,22 +35,44 @@
...
-	 * @throws JavaModelException
+	 * 
 	 * @throws CoreException
 	 */
-	void collectDiagnostics(JavaDiagnosticsContext context, IProgressMonitor monitor)
-			throws JavaModelException, CoreException;
+	void collectDiagnostics(JavaDiagnosticsContext context, IProgressMonitor monitor) throws CoreException;
 
+	/**
+	 * End the collection of diagnostics.
```
Comment (xorye) [inline microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/java/IJavaDiagnosticsParticipant.java]: End diagnostics collection.

Comment (angelozerr) [reply microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/java/IJavaDiagnosticsParticipant.java]: fixed.

File: microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/java/IJavaDiagnosticsParticipant.java (hunk 4 above)
Comment (angelozerr) [reply microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/java/IJavaDiagnosticsParticipant.java]: fixed.

File: microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/java/IJavaHoverParticipant.java (hunk 3 above)
Comment (angelozerr) [reply microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/java/IJavaHoverParticipant.java]: fixed.

File: microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/java/IJavaHoverParticipant.java (hunk 2 above)
Comment (angelozerr) [reply microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/java/IJavaHoverParticipant.java]: fixed.

File: microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/java/IJavaDiagnosticsParticipant.java (hunk 1 above)
Comment (angelozerr) [reply microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/java/IJavaDiagnosticsParticipant.java]: When I move LSP4XML to Eclipse.org, this information missed. 

@fbricon we need to add this information, no?

Comment (angelozerr) [reply microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/java/IJavaDiagnosticsParticipant.java]: @xorye yes we need this header.

**Comment by xorye** [comment]
Awesome, I'd say this is good to merge

**Comment by angelozerr** [comment]
Thanks for your review @xorye !

**Pull request merged**